core/src/main/scala-2/caliban/schema/SchemaDerivation.scala

```diff
@@ -231,20 +231,28 @@ trait SchemaDerivation[R] extends LowPriorityDerivedSchema {
 
   private def getDescription[Typeclass[_], Type](ctx: ReadOnlyParam[Typeclass, Type]): Option[String] =
     getDescription(ctx.annotations)
+}
+
+trait SchemaDerivation[R] extends CommonSchemaDerivation[R] {
 
   /**
-   * Generates an instance of `Schema` for the given type T.
-   * This should be used only if T is a case class or a sealed trait.
+   * Returns an instance of `Schema` for the given type T.
+   * This method requires a `Schema` for all types nested inside T.
+   * It should be used only if T is a case class or a sealed trait.
    */
-  implicit def genMacro[T]: Derived[Typeclass[T]] = macro DerivedMagnolia.derivedMagnolia[Typeclass, T]
+  def gen[R0, T]: Typeclass[T] = macro Magnolia.gen[T]
 
   /**
    * Returns an instance of `Schema` for the given type T.
-   * For a case class or sealed trait, you can call `genMacro[T].schema` instead to get more details if the
-   * schema can't be derived.
+   * This method will automatically generate missing `Schema` for all types nested inside T that are case classes or sealed traits.
    */
-  def gen[R0, T](implicit derived: Derived[Schema[R0, T]]): Schema[R0, T] = derived.schema
+  def genAll[R0, T](implicit derived: Derived[Schema[R0, T]]): Schema[R0, T] = derived.schema
 
+  object auto extends AutoSchemaDerivation[R]
```

Comment:
I think this encourages the user to use this mode, what about not adding it here?

Reply:
where do you suggest them?

Reply:
In the `AutoSchemaDerivation` trait maybe? But not in the non-automatic mode

Reply:
Sorry to come back on this but why do you want to add this `auto` thing in this `SchemaDerivation` trait? 🤔

Reply:
I want to enabled auto-derivation using this import: `import caliban.schema.Schema.auto._`

Reply:
Why not put this object into the `Schema` object then? 🤔

Reply:
Because when `R` is not `Any` in Scala 2, you need to extend `GenericSchema[R]` and import `auto._` from that object. `Schema` is fixed to `Any`.